packages/shared/src/components/post/write/ShareLink.tsx

```diff
@@ -38,18 +40,20 @@ export function ShareLink({
   moderated,
 }: ShareLinkProps): ReactElement {
```

Comment:
This component is in charge for both creation and update.
I moved the creation to the new logic, kept the old one for update